pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -390,25 +390,14 @@ func (c *Controller) addNodeRoute(nodeName string, node *corev1.Node) error {
 		return nil
 	}
 
-	klog.Infof("Adding routes and flows to Node %s, podCIDR: %s, addresses: %v",
-		nodeName, node.Spec.PodCIDR, node.Status.Addresses)
+	klog.Infof("Adding routes and flows to Node %s, podCIDRs: %v, addresses: %v",
+		nodeName, node.Spec.PodCIDRs, node.Status.Addresses)
 
-	if node.Spec.PodCIDR == "" {
-		klog.Errorf("PodCIDR is empty for Node %s", nodeName)
-		// Does not help to return an error and trigger controller retries.
-		return nil
-	}
-	peerPodCIDRAddr, peerPodCIDR, err := net.ParseCIDR(node.Spec.PodCIDR)
-	if err != nil {
-		klog.Errorf("Failed to parse PodCIDR %s for Node %s", node.Spec.PodCIDR, nodeName)
-		return nil
-	}
 	peerNodeIP, err := GetNodeAddr(node)
```

Comment:
Maybe we should add some comments regarding peerNodeIP? Based on our tests and the open issue https://github.com/kubernetes/kubernetes/issues/91940, there is only a single InternalIP for now. It will also decide IPv4/IPv6 tunnel we will use and it is IPv4 only at this stage.

Reply:
Make sense, I will add comments to explain it.